app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt

```diff
@@ -163,6 +163,20 @@ class MessageActivity : AppCompatActivity() {
 
         drawableRecyclerAdapter = DrawableAdapter(this, listOfDrawables)
         drawableRecyclerView.adapter = drawableRecyclerAdapter
+
+        val drawableClickListener = object : DrawableAdapter.DrawableClickListener {
+            override fun onClick(position: Int) {
+                drawableRecyclerAdapter.setSelectedItemPosition(position)
+                presenter.subscribeCurrentDrawable(position)
+                drawableRecyclerAdapter.notifyDataSetChanged()
```

Comment:
```suggestion
```

Reply:
To be honest, I'm not sure if moving those methods into the Adapter class would be a better change... it's like splitting the function into two for me. When I'm reading the code inside the MessageActivity class, I can immediately understand what the click listener do. Following your suggestions would make me only get the "OK, the click listener would save the drawable position to the presenter, but that's all". Please let me know what you think

Reply:
When someone is reading the MessageActivity, I guess he should only get to know the functions that are essential. selecting the item and notifying the adapter are irrelevant as they have to be done everytime anyways, so why not just include the functions inside the adapter itself. It would make the code clean and easy to read the onClick function

Reply:
I think we can wait for some comments from the maintainer of this project on this discussion

Reply:
@anhanh11001 This should be handled in Preview custom view I think. Similar to how EditText in Android handle rotation without any developer intervention

Reply:
I think you misunderstood the PR, it is used to retain the picked drawable in the adapter and recycler view. This PR doesn't help retain the animation in the custom Preview view